draft-ietf-masque-h3-datagram.md

```diff
@@ -196,7 +196,7 @@ client-initiated bidirectional streams, and those have stream IDs that are
 divisible by four). The largest legal QUIC stream ID value is 2<sup>62</sup>-1,
 so the largest legal value of Quarter Stream ID is 2<sup>62</sup>-1 / 4.
 Receipt of a frame that includes a larger value MUST be treated as a connection
-error of type FRAME_ENCODING_ERROR.
+error of type H3_FRAME_ERROR.
```

Comment:
This is a funny one.

HTTP/3 Datagram is not an HTTP/3 frame, so reporting this error code feels odd. 

I think it was me that picked FRAME_ENCODING_ERROR, which is defined in transport

> FRAME_ENCODING_ERROR (0x07):  An endpoint received a frame that was
      badly formatted -- for instance, a frame of an unknown type or an
      ACK frame that has more acknowledgment ranges than the remainder
      of the packet could carry.

But really, this is not a problem in the transport layer, and that also feels wrong now I think about it. I think the answer is a new application-layer `H3_DATAGRAM_ERROR`.

Reply:
I totally agree that borrowing error codes from lower level protocols is weird.  

H3_DATAGRAM_ERROR works when processing HTTP/3 DATAGRAMs.  But the capsule length errors don't fit with that.  Does Capsule Protocol needs its own error code range too (it can be conveyed over non-H3)?

Reply:
I agree these are two different types of error